src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,145 @@
+        if (size == 0 && index == 0) {
+            nodeNew = new Node<>(null, value, null);
+            first = last = nodeNew;
+            size++;
+            return;
+        }
+        if (index == size) {
+            nodeNew = new Node<>(last, value, null);
+            last.next = nodeNew;
+            last = nodeNew;
+            size++;
+            return;
+        }
+        isValid(index);
+        nodeTemp = getNode(index);
+        if (nodeTemp == first) {
+            nodeNew = new Node<>(null, value, nodeTemp);
+            nodeTemp.previous = nodeNew;
+            first = nodeNew;
+            size++;
+            return;
+        } else {
+            nodeNew = new Node<>(nodeTemp.previous, value, nodeTemp);
+            nodeTemp.previous.next = nodeNew;
+            nodeTemp.previous = nodeNew;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T object : list) {
+            add(object);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        nodeTemp = getNode(index);
+        T value1 = nodeTemp.value;
+        nodeTemp.value = value;
+        return value1;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        nodeTemp = getNode(index);
+        deletedNode(nodeTemp);
+        return nodeTemp.value;
     }
 
     @Override
-    public boolean remove(T object) {
+   public boolean remove(T object) {
+        nodeTemp = first;
+        while (nodeTemp != null) {
+            if (nodeTemp.value == null || nodeTemp.value.equals(object)) {
+                deletedNode(nodeTemp);
+                return true;
+            }
+            nodeTemp = nodeTemp.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private Node<T> previous;
+        private T value;
+        private Node<T> next;
+
+        Node(Node<T> previous, T value, Node<T> next) {
+            this.previous = previous;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        isValid(index);
+        if (index < size / 2) {
+            nodeTemp = first;
+            for (int i = 0; i < index; i++) {
+                nodeTemp = nodeTemp.next;
+            }
+        } else {
+            nodeTemp = last;
+            for (int i = size - 1; i > index; i--) {
+                nodeTemp = nodeTemp.previous;
+            }
+        }
+        return nodeTemp;
+    }
+
+    private void deletedNode(Node<T> node) {
+        if (node == last) {
```

Comment:
cane we compare 2 objects with '=='
let's compare node.prev == null or node.next == null to know if it's tail or head